internal/restic/node_windows.go

```diff
@@ -407,38 +407,75 @@ func (node *Node) fillGenericAttributes(path string, fi os.FileInfo, stat *statT
-	// Check if it's an extended length path
-	if strings.HasPrefix(path, uncPathPrefix) {
-		// Convert \\?\UNC\ extended path to standard path to get the volume name correctly
-		path = `\\` + path[len(uncPathPrefix):]
-	} else if strings.HasPrefix(path, extendedPathPrefix) {
-		//Extended length path prefix needs to be trimmed to get the volume name correctly
-		path = path[len(extendedPathPrefix):]
-	} else if strings.HasPrefix(path, globalRootPrefix) {
-		// EAs are not supported for \\?\GLOBALROOT i.e. VSS snapshots
-		return false, nil
-	} else {
-		// Use the absolute path
-		path, err = filepath.Abs(path)
-		if err != nil {
-			return false, fmt.Errorf("failed to get absolute path: %w", err)
-		}
+	var volumeName string
+	volumeName, err = prepareVolumeName(path)
+	if err != nil {
+		return false, err
 	}
-	volumeName := filepath.VolumeName(path)
-	if volumeName == "" {
-		return false, nil
+
+	if volumeName != "" {
+		// First check if the manually prepared volume name is already in the map
+		eaSupportedValue, exists := eaSupportedVolumesMap.Load(volumeName)
+		if exists {
+			// Cache hit, immediately return the cached value
+			return eaSupportedValue.(bool), nil
+		}
+		// If not found, check if EA is supported with manually prepared volume name
+		isEASupportedVolume, err = fs.PathSupportsExtendedAttributes(volumeName + `\`)
+		// If the prepared volume name is not valid, we will next fetch the actual volume name.
+		if err != nil && !errors.Is(err, windows.DNS_ERROR_INVALID_NAME) {
+			return false, err
+		}
 	}
-	eaSupportedValue, exists := eaSupportedVolumesMap.Load(volumeName)
-	if exists {
-		return eaSupportedValue.(bool), nil
+	// If an entry is not found, get the actual volume name using the GetVolumePathName function
+	volumeNameActual, err := fs.GetVolumePathName(path)
+	if err != nil {
+		return false, err
+	}
+	if volumeNameActual != volumeName {
+		// If the actual volume name is different, check cache for the actual volume name
+		eaSupportedValue, exists := eaSupportedVolumesMap.Load(volumeNameActual)
+		if exists {
+			// Cache hit, immediately return the cached value
+			return eaSupportedValue.(bool), nil
+		}
+		// If the actual volume name is different and is not in the map, again check if the new volume supports extended attributes with the actual volume name
+		isEASupportedVolume, err = fs.PathSupportsExtendedAttributes(volumeNameActual + `\`)
+		if err != nil {
+			return false, err
+		}
 	}
+	eaSupportedVolumesMap.Store(volumeNameActual, isEASupportedVolume)
+	return isEASupportedVolume, err
+}
 
-	// Add backslash to the volume name to ensure it is a valid path
-	isEASupportedVolume, err = fs.PathSupportsExtendedAttributes(volumeName + `\`)
-	if err == nil {
-		eaSupportedVolumesMap.Store(volumeName, isEASupportedVolume)
+// prepareVolumeName prepares the volume name for different cases in Windows
+func prepareVolumeName(path string) (volumeName string, err error) {
```

Comment:
Please add a minimal test for this function.

Reply:
done